sql/core/src/main/scala/org/apache/spark/sql/execution/command/cache.scala

```diff
@@ -19,26 +19,27 @@ package org.apache.spark.sql.execution.command
 
 import java.util.Locale
 
-import org.apache.spark.sql.{Dataset, Row, SparkSession}
-import org.apache.spark.sql.catalyst.TableIdentifier
+import org.apache.spark.sql.{AnalysisException, DataFrame, Dataset, Row, SparkSession}
 import org.apache.spark.sql.catalyst.plans.QueryPlan
 import org.apache.spark.sql.catalyst.plans.logical.{IgnoreCachedData, LogicalPlan}
 import org.apache.spark.sql.catalyst.util.CaseInsensitiveMap
+import org.apache.spark.sql.connector.catalog.CatalogV2Implicits.MultipartIdentifierHelper
 import org.apache.spark.storage.StorageLevel
 
 case class CacheTableCommand(
```

Comment:
the next thing we can do is to refactor it using the v2 framework (not adding a v2 version). The benefits are: 1. moving the logical plan to catalyst. 2. resolve the table in the analyzer. e.g.
```
CacheTable(UnresolvedRelation(...), ...)
...
case class CacheTableExec(relation: LogicalPlan) {
  def run() {
     val df  = Dataset.ofRows(spark, relation)
     ....
  }
}
```

Reply:
OK, will do.

Reply:
One issue I am encountering by moving to the v2 framework (for v2 tables) is the following.

When `CACHE TABLE testcat.tbl` is run, `tbl` is changed from `DataSourceV2Relation` to `DataSourceV2ScanRelation` in `V2ScanRelationPushDown` rule, now that the plan goes thru analyzer, optimizer, etc. But, if I run `spark.table("testcat.tbl")`, the query execution has `tbl` as `DataSourceV2Relation`, thus cache is not applied.

Reply:
ah, one solution is to follow `InsertIntoStatement` and do not make the `table` as a child. Then we resolve `UnresolvedRelation` inside `CacheTable` manually in `ResolveTempViews` and other resolution rules.